src/WebUI/Options/OpenIdConnectPostConfigureOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Codidact.Application.Common.Interfaces;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.Options;
+
+namespace Codidact.WebUI.Options
+{
+    public class OpenIdConnectPostConfigureOptions : IPostConfigureOptions<OpenIdConnectOptions>
+    {
+        private readonly ISecretsService _secretsService;
+
+        public OpenIdConnectPostConfigureOptions(ISecretsService secretsService)
+        {
+            _secretsService = secretsService;
+        }
+
+        public async void PostConfigure(string name, OpenIdConnectOptions options)
```

Comment:
`PostConfigure` is called synchronously, so when the secrets service implementation actually runs asynchronously (which the `DevelopmentSecretsService` does not do), then this will not work.

Reply:
Switched to .GetAwaiter().GetResult() to make it sync.